### ISSUE TITLE
MySQL: "Specified key was too long" on `doctrine:create`

### ISSUE THREAD

**Issue by ComiR** [neos/neos-development-collection#2168]
If you use MySQLs InnoDB and `utf8mb4` the [defined index of `NodeData`'s `parentpath`](https://github.com/neos/neos-development-collection/blob/master/Neos.ContentRepository/Classes/Domain/Model/NodeData.php#L42) is too long. The maximum key length is 3072 bytes, which in `utf8mb4` means 768 chars. The length of `parentpath` is [defined](https://github.com/neos/neos-development-collection/blob/master/Neos.ContentRepository/Classes/Domain/Model/NodeData.php#L94) as 4000 chars.

**Comment by kitsunet** [comment]
Which MySQL version is that? Because we specifically raised the requirement for Neos 4.0 to avoid key length problems.
https://www.neos.io/download-and-extend/upgrade-instructions-3-3-4-0.html

**Comment by ComiR** [comment]
Yeah, I know. The limit specifically applies to the correctly configured database. Otherwise it would be 767 (or 787) Bytes.
See documentation for [MariaDB](https://mariadb.com/kb/en/library/innodb-limitations/#large-prefix-size) or [MySQL](https://dev.mysql.com/doc/refman/8.0/en/innodb-restrictions.html#innodb-maximums-minimums).

**Comment by kdambekalns** [comment]
I have not heard about this from someone else, so I beg to differ. Also, does this happen with `doctrine:migrate` for you? If it does, this seems to be an issue with `doctrine:create` (which should not really be used, ever.)

**Comment by ComiR** [comment]
I couldn't reproduce it with current 4.0 and master branches.
I currently only tested it with MariaDB 10.1.35 (not officially supported).

Maybe an update of doctrine solved the problem. If I execute the SQL by hand (`./flow doctrine:create --output create.sql` relevant line below), I still get the warning.

<details>
<summary><pre>neos_contentrepository_domain_model_nodedata</pre></summary>

```sql
CREATE TABLE neos_contentrepository_domain_model_nodedata
    (
        persistence_object_identifier VARCHAR(40) NOT NULL,
        workspace VARCHAR(255) DEFAULT NULL,
        contentobjectproxy VARCHAR(40) DEFAULT NULL,
        movedto VARCHAR(40) DEFAULT NULL,
        version INT DEFAULT 1 NOT NULL,
        pathhash VARCHAR(32) NOT NULL,
        path VARCHAR(4000) NOT NULL,
        parentpathhash VARCHAR(32) NOT NULL,
        parentpath VARCHAR(4000) NOT NULL,
        identifier VARCHAR(255) NOT NULL,
        sortingindex INT DEFAULT NULL,
        removed TINYINT(1) NOT NULL,
        dimensionshash VARCHAR(32) NOT NULL,
        lastmodificationdatetime DATETIME NOT NULL,
        lastpublicationdatetime DATETIME DEFAULT NULL,
        hiddenbeforedatetime DATETIME DEFAULT NULL,
        hiddenafterdatetime DATETIME DEFAULT NULL,
        dimensionvalues LONGTEXT NOT NULL COMMENT '(DC2Type:flow_json_array)',
        properties LONGTEXT NOT NULL COMMENT '(DC2Type:flow_json_array)',
        nodetype VARCHAR(255) NOT NULL,
        creationdatetime DATETIME NOT NULL,
        hidden TINYINT(1) NOT NULL,
        hiddeninindex TINYINT(1) NOT NULL,
        accessroles LONGTEXT NOT NULL COMMENT '(DC2Type:flow_json_array)',
        INDEX IDX_CE6515698D940019 (workspace),
        INDEX IDX_CE6515694930C33C (contentobjectproxy),
        UNIQUE INDEX UNIQ_CE6515692D45FE4D (movedto),
        INDEX parentpath_sortingindex (parentpathhash, sortingindex),
        INDEX parentpath (parentpath),
        INDEX identifierindex (identifier),
        INDEX nodetypeindex (nodetype),
        UNIQUE INDEX UNIQ_CE6515692DBEC7578D94001992F8FB01 (pathhash, workspace, dimensionshash),
        UNIQUE INDEX UNIQ_CE651569772E836A8D94001992F8FB012D45FE4D (identifier, workspace, dimensionshash, movedto),
        PRIMARY KEY(persistence_object_identifier)
    )
    DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE = INNODB
;
```
</details>

**Comment by kdambekalns** [comment]
Well, 10.2 differs from 10.1, guess why… ;)

`innodb_large_prefix` has been deprecated/removed, that is, no length restrictions are in place anymore, as far as I understood it.

**Comment by ComiR** [comment]
Restriction is still the same, just ON by default ;)
![image](https://user-images.githubusercontent.com/11410095/47733975-b6b82b80-dc69-11e8-8b2c-a2769bb8a043.png)

**Comment by kdambekalns** [comment]
I just ran this successfully on MariaDB 10.2.18, no errors whatsoever:

```
CREATE TABLE `neos_contentrepository_domain_model_nodedata` (
  `persistence_object_identifier` varchar(40) COLLATE utf8mb4_unicode_ci NOT NULL,
  `workspace` varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `contentobjectproxy` varchar(40) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `path` varchar(4000) COLLATE utf8mb4_unicode_ci NOT NULL,
  `identifier` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
  `sortingindex` int(11) DEFAULT NULL,
  `properties` longtext COLLATE utf8mb4_unicode_ci NOT NULL COMMENT '(DC2Type:flow_json_array)',
  `nodetype` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
  `removed` tinyint(1) NOT NULL,
  `hidden` tinyint(1) NOT NULL,
  `hiddenbeforedatetime` datetime DEFAULT NULL,
  `hiddenafterdatetime` datetime DEFAULT NULL,
  `hiddeninindex` tinyint(1) NOT NULL,
  `accessroles` longtext COLLATE utf8mb4_unicode_ci NOT NULL COMMENT '(DC2Type:flow_json_array)',
  `version` int(11) NOT NULL DEFAULT 1,
  `parentpath` varchar(4000) COLLATE utf8mb4_unicode_ci NOT NULL,
  `pathhash` varchar(32) COLLATE utf8mb4_unicode_ci NOT NULL,
  `dimensionshash` varchar(32) COLLATE utf8mb4_unicode_ci NOT NULL,
  `dimensionvalues` longtext COLLATE utf8mb4_unicode_ci NOT NULL COMMENT '(DC2Type:flow_json_array)',
  `parentpathhash` varchar(32) COLLATE utf8mb4_unicode_ci NOT NULL,
  `movedto` varchar(40) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `creationdatetime` datetime NOT NULL,
  `lastmodificationdatetime` datetime NOT NULL,
  `lastpublicationdatetime` datetime DEFAULT NULL,
  PRIMARY KEY (`persistence_object_identifier`),
  UNIQUE KEY `UNIQ_CE6515692DBEC7578D94001992F8FB01` (`pathhash`,`workspace`,`dimensionshash`),
  UNIQUE KEY `UNIQ_CE651569772E836A8D94001992F8FB012D45FE4D` (`identifier`,`workspace`,`dimensionshash`,`movedto`),
  KEY `parentpath_sortingindex` (`parentpathhash`,`sortingindex`),
  KEY `identifierindex` (`identifier`),
  KEY `nodetypeindex` (`nodetype`),
  KEY `IDX_CE6515698D940019` (`workspace`),
  KEY `IDX_CE6515694930C33C` (`contentobjectproxy`),
  KEY `IDX_CE6515692D45FE4D` (`movedto`),
  KEY `parentpath` (`parentpath`(255)),
  CONSTRAINT `FK_60A956B92D45FE4D` FOREIGN KEY (`movedto`) REFERENCES `neos_contentrepository_domain_model_nodedata` (`persistence_object_identifier`) ON DELETE SET NULL,
  CONSTRAINT `FK_60A956B98D940019` FOREIGN KEY (`workspace`) REFERENCES `neos_contentrepository_domain_model_workspace` (`name`) ON DELETE SET NULL,
  CONSTRAINT `neos_contentrepository_domain_model_nodedata_ibfk_2` FOREIGN KEY (`contentobjectproxy`) REFERENCES `neos_contentrepository_domain_model_contentobjectproxy` (`persistence_object_identifier`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```

**Comment by kdambekalns** [comment]
As I wrote in https://discuss.neos.io/t/rfc-switch-to-utf8mb4-raise-db-requirements/3274

> Can we raise DB requirements, please? For this to work we need at least MySQL 5.7.7 or MariaDB 10.2.2 to avoid key length issues with the new charset. MariaDB 10.2.2 was released on 2016-09-27, MySQL v5.7.7 on 2015-04-08).

**Comment by gerhard-boden** [comment]
The requirement check in the setup should warn you if your Maria DB Version is below 10.2:
https://github.com/neos/setup/pull/48/files
